qiskit/circuit/equivalence.py

```diff
@@ -12,190 +12,24 @@
 
 """Gate equivalence library."""
 
-import copy
-from collections import namedtuple
-
 from rustworkx.visualization import graphviz_draw
 import rustworkx as rx
 
-from qiskit.exceptions import InvalidFileError
-from .exceptions import CircuitError
-from .parameter import Parameter
-from .parameterexpression import ParameterExpression
 
-Key = namedtuple("Key", ["name", "num_qubits"])
-Equivalence = namedtuple("Equivalence", ["params", "circuit"])  # Ordered to match Gate.params
-NodeData = namedtuple("NodeData", ["key", "equivs"])
-EdgeData = namedtuple("EdgeData", ["index", "num_gates", "rule", "source"])
+from qiskit.exceptions import InvalidFileError
+from qiskit._accelerate.equivalence import (  # pylint: disable=unused-import
+    BaseEquivalenceLibrary,
+    Key,
+    Equivalence,
+    NodeData,
```

Comment:
Some things, eg. `Key` and `NodeData`, are not used apparently, and the lint for this is suppressed. Why is this?

Reply:
This is to allow the user to still be able to import these things directly from `qiskit.circuit.equivalence` as they were able to before. If this is not needed anymore I can easily remove.

Reply:
Oh sorry, I did not check who calls these. `Key` and `NodeData` are called from Python in `class BasisTranslator`. Looks like they cannot be removed after all.